mw_auth_key.go

```diff
@@ -2,12 +2,19 @@ package main
 
 import (
 	"errors"
+	"fmt"
 	"net/http"
 	"strings"
 
 	"github.com/TykTechnologies/tyk/apidef"
 	"github.com/TykTechnologies/tyk/certs"
 	"github.com/TykTechnologies/tyk/request"
+	"github.com/TykTechnologies/tyk/signature_validator"
+)
+
+const (
+	defaultErrorCode    = http.StatusForbidden
```

Comment:
Default should be `http.StatusUnauthorized`.

Because if signature fails, we do not believe the user is who they say that they are - even if they have correct auth token.